### PR TITLE
Allow setting device flags in the metadata

### PR DESCRIPTION
**Pull request by hughsie**
NOTE: This can only be used to affect device write, not device enumeration.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
This needs a lot of supporting LVFS code too, and a lot of server-side validation on each of the valid flags for each protocol.

**Comment by hughsie** [comment]
LVFS side: https://gitlab.com/fwupd/lvfs-website/-/merge_requests/1403

File: libfwupdplugin/fu-device.c [hunk 1]
```diff
@@ -5519,6 +5534,8 @@ fu_device_ensure_from_component(FuDevice *self, XbNode *component)
 		fu_device_ensure_from_component_signed(self, component);
 	if (fu_device_has_internal_flag(self, FU_DEVICE_INTERNAL_FLAG_MD_SET_VERFMT))
 		fu_device_ensure_from_component_verfmt(self, component);
+	if (fu_device_has_internal_flag(self, FU_DEVICE_INTERNAL_FLAG_MD_SET_FLAGS))
```
Comment (superm1) [inline libfwupdplugin/fu-device.c:5537]: I guess you're missing to add this flag to NVME plugin still

**Review by superm1 — APPROVED** [review]
approach looks good to me

File: libfwupdplugin/fu-device.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-device.c:5537]: Yup, will do in the other PR.

**Pull request merged**